docker/Dockerfile

```diff
@@ -1,65 +1,62 @@
-# 'builder' includes all build & testing dependencies.
-FROM debian:12 AS builder
+# 'fenv-builder' extends the fenv base image with FQL build & testing dependencies.
+# This stage is used by fenv for local development and CI/CD verification tasks.
+ARG FENV_DOCKER_TAG
+FROM fenv:${FENV_DOCKER_TAG} AS fenv-builder
 
-RUN apt-get update &&\
+# Install additional system packages needed for build
+RUN apt-get update && \
     apt-get install --no-install-recommends -y \
       build-essential=12.9 \
       ca-certificates=20* \
       git=1:2.39.* \
-      curl=7.88.* &&\
-    apt-get clean &&\
+      curl=7.88.* && \
+    apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
-# The FDB installer is not deleted so it can be used to
-# install the library in the final stage at the end of
-# this file.
-ARG FDB_LIB_URL
-RUN curl -Lo /fdb.deb $FDB_LIB_URL &&\
-    dpkg -i /fdb.deb
-
-ARG GO_URL
-RUN curl -Lo /go.tar.gz $GO_URL &&\
-    tar -C /usr/local -xzf /go.tar.gz &&\
-    rm /go.tar.gz
+# Install Go
+ARG GO_URL="https://go.dev/dl/go1.19.1.linux-amd64.tar.gz"
+RUN curl -fsSL ${GO_URL} | tar -C /usr/local -xz
 ENV PATH="/root/go/bin:/usr/local/go/bin:${PATH}"
 ENV GOCACHE="/cache/gocache"
 ENV GOMODCACHE="/cache/gomod"
 
+# Install golangci-lint
 ARG GOLANGCI_LINT_URL="https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh"
-ARG GOLANGCI_LINT_VER
-RUN curl -Lo /golint.sh $GOLANGCI_LINT_URL &&\
-    sh /golint.sh -b "$(go env GOPATH)/bin" $GOLANGCI_LINT_VER &&\
-    rm /golint.sh
+ARG GOLANGCI_LINT_VER="v1.49.0"
+RUN curl -fsSL ${GOLANGCI_LINT_URL} | sh -s -- -b "$(go env GOPATH)/bin" ${GOLANGCI_LINT_VER}
 ENV GOLANGCI_LINT_CACHE="/cache/golangci-lint"
 
-ARG SHELLCHECK_URL
-RUN curl -Lo /shellcheck.tar.xz $SHELLCHECK_URL &&\
-    tar -xf /shellcheck.tar.xz &&\
-    mv /shellcheck-*/shellcheck /usr/local/bin &&\
-    rm -r /shellcheck.tar.xz /shellcheck-*
+# Install shellcheck
+ARG SHELLCHECK_URL="https://github.com/koalaman/shellcheck/releases/download/v0.10.0/shellcheck-v0.10.0.linux.x86_64.tar.xz"
+RUN curl -fsSL ${SHELLCHECK_URL} -o /tmp/shellcheck.tar.xz && \
+    tar -xf /tmp/shellcheck.tar.xz -C /tmp && \
+    mv /tmp/shellcheck-*/shellcheck /usr/local/bin && \
+    rm -rf /tmp/shellcheck*
 
-ARG HADOLINT_URL
-RUN curl -Lo /usr/local/bin/hadolint $HADOLINT_URL &&\
+# Install hadolint
+ARG HADOLINT_URL="https://github.com/hadolint/hadolint/releases/download/v2.7.0/hadolint-Linux-x86_64"
+RUN curl -fsSL ${HADOLINT_URL} -o /usr/local/bin/hadolint && \
     chmod +x /usr/local/bin/hadolint
 
-ARG JQ_URL
-RUN curl -Lo /usr/local/bin/jq $JQ_URL &&\
+# Install jq
+ARG JQ_URL="https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64"
+RUN curl -fsSL ${JQ_URL} -o /usr/local/bin/jq && \
     chmod +x /usr/local/bin/jq
```

Comment:
We don't need `jq`. We only required it before for initializing the database, and `fenv` does this for us already.